truss/remote/baseten/api.py

```diff
@@ -570,6 +570,22 @@ def list_training_jobs(self, project_id: str):
         resp_json = self._rest_api_client.get(f"v1/training_projects/{project_id}/jobs")
         return resp_json
 
+    def list_all_training_jobs(
+        self,
+        status_filter: Optional[List[str]] = None,
```

Comment:
If you like the `search` suggestion, I think this could just be `statuses`, since the `_filter` is implied by the `search` target